projects/components/src/table/data/table-cdk-data-source-api.ts

```diff
@@ -5,6 +5,7 @@ import { TableDataSource } from './table-data-source';
 
 export interface TableDataSourceProvider {
   data?: TableDataSource<TableRow>;
+  mode?: TableMode;
```

Comment:
I thought the table controls change was all meant to avoid changing table to support the flattened use case.This appears to read like we're giving the mode to the cdk data source just so it can spit it back out with the request - but why is that necessary? Whatever is receiving the request presumably knows whether it's flattened or not (to provide this in the first place)